ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -7705,6 +7705,13 @@ protected Operator genFileSinkPlan(String dest, QB qb, Operator input)
       } else {
         tableDescriptor = PlanUtils.getTableDesc(tblDesc, cols, colTypes);
       }
+
+      // if available, set location in table desc properties
+      if (tblDesc != null && tblDesc.getLocation() != null && tableDescriptor != null &&
+          !tableDescriptor.getProperties().containsKey(hive_metastoreConstants.META_TABLE_LOCATION)) {
```

Comment:
how does this work with the HMSTranslators? (These are the stuff which makes sure that external/managed table is put into the correct location)

Reply:
If the location clause is present in the query, the translator should not touch that.